HiFiasm for T2T Tuli and Wagyu assembly

Data are here: /hpcfs/groups/phoenix-hpc-avsci/Lloyd_Low/Tuli_x_Wagyu_data/assembly_contig/hifiasm/TxWHiFiTrioT2T_DoraONT40k

Wagyu initial contig stats:

| chr | contig | query_name                                      | query_length | ref_length | contig_align_proportion       | queryref_proportion | queryref_proportion_check |
|-----|--------|-------------------------------------------------|--------------|------------|-------------------------------|---------------------|---------------------------|
| 1   | 1      | h2tg000026l                                     | 166872203    | 158534110  | 105.2595                      | 105.2595            | 105.259494628632          |
| 2   | 1      | h2tg000012l                                     | 138096131    | 136231102  | 101.369                       | 101.369             | 101.369018507976          |
| 3   | 1      | h2tg000024l                                     | 113161246    | 121005158  | 93.5177                       | 93.5177             | 93.5177044271121          |
| 4   | 1      | h2tg000003l                                     | 130003985    | 120000601  | 108.3361                      | 108.3361            | 108.336111583308          |
| 5   | 1      | h2tg000019l                                     | 132489435    | 120089316  | 110.3257                      | 110.3257            | 110.325747046473          |
| 6   | 1      | h2tg000021l                                     | 113518087    | 117806340  | 96.3599                       | 96.3599             | 96.359913227081           |
| 7   | 2      | h2tg000009l,h2tg000269l                         | 117475077    | 110682743  | 105.9777,0.159                | 106.1367            | 106.136759729563          |
| 8   | 2      | h2tg000020l,h2tg000057l                         | 121983410    | 113319770  | 93.135,14.5103                | 107.6453            | 107.645303198197          |
| 9   | 1      | h2tg000006l                                     | 116313833    | 105454467  | 110.2977                      | 110.2977            | 110.297682316293          |
| 10  | 1      | h2tg000044l                                     | 110093730    | 103308737  | 106.5677                      | 106.5677            | 106.567685557902          |
| 11  | 1      | h2tg000011l                                     | 114533536    | 106982474  | 107.0582                      | 107.0582            | 107.058223387132          |
| 12  | 1      | h2tg000002l                                     | 102068089    | 87216183   | 117.0288                      | 117.0288            | 117.028841998279          |
| 13  | 2      | h2tg000001l,h2tg000053l                         | 89741809     | 83472345   | 93.778,13.7328                | 107.5108            | 107.510827687901          |
| 14  | 1      | h2tg000049l                                     | 83630686     | 82403003   | 101.4899                      | 101.4899            | 101.489852256962          |
| 15  | 1      | h2tg000007l                                     | 91998830     | 85007780   | 108.224                       | 108.224             | 108.224011966905          |
| 16  | 1      | h2tg000005l                                     | 91180055     | 81013979   | 112.5485                      | 112.5485            | 112.548545480034          |
| 17  | 2      | h2tg000008l,h2tg000040l                         | 81425041     | 73167244   | 30.6665,80.6197               | 111.2862            | 111.286193860192          |
| 18  | 2      | h2tg000016l,h2tg000023l                         | 71274640     | 65820629   | 84.4553,23.8309               | 108.2862            | 108.286172713421          |
| 19  | 3      | h2tg000033l,h2tg000079l,h2tg000076l             | 63048937     | 63449741   | 68.0874,2.9935,28.2873        | 99.3682             | 99.3683126303069          |
| 20  | 1      | h2tg000043l                                     | 72127045     | 71974595   | 100.2118                      | 100.2118            | 100.21181084798           |
| 21  | 1      | h2tg000032l                                     | 86688117     | 69862954   | 124.0831                      | 124.0831            | 124.083097030223          |
| 22  | 1      | h2tg000035l                                     | 61228083     | 60773035   | 100.7488                      | 100.7488            | 100.748766290839          |
| 23  | 1      | h2tg000031l                                     | 54566597     | 52498615   | 103.9391                      | 103.9391            | 103.939117250998          |
| 24  | 4      | h2tg000110l,h2tg000098l,h2tg000072l,h2tg000015l | 59251632     | 62317253   | 19.5922,5.8194,4.1686,65.5004 | 95.0806             | 95.0806223759574          |
| 25  | 1      | h2tg000014l                                     | 48243632     | 42350435   | 113.9153                      | 113.9153            | 113.915316336184          |
| 26  | 1      | h2tg000059l                                     | 55674231     | 51992305   | 107.0817                      | 107.0817            | 107.081674874772          |
| 27  | 1      | h2tg000022l                                     | 52298086     | 45612108   | 114.6583                      | 114.6583            | 114.65834028105           |
| 28  | 1      | h2tg000028l                                     | 43217833     | 45940150   | 94.0742                       | 94.0742             | 94.0742095966165          |
| 29  | 1      | h2tg000013l                                     | 54041122     | 51098607   | 105.7585                      | 105.7585            | 105.758503358027          |

Tuli initial contig stats:

| chr | contig | query_name              | query_length | ref_length | contig_align_proportion | queryref_proportion | queryref_proportion_check |
|-----|--------|-------------------------|--------------|------------|-------------------------|---------------------|---------------------------|
| 1   | 1      | h1tg000003l             | 166683787    | 158534110  | 105.1406                | 105.1406            | 105.14064575756           |
| 2   | 1      | h1tg000017l             | 153656481    | 136231102  | 112.791                 | 112.791             | 112.791043120241          |
| 3   | 1      | h1tg000010l             | 113206779    | 121005158  | 93.5553                 | 93.5553             | 93.5553334015729          |
| 4   | 1      | h1tg000007l             | 121629509    | 120000601  | 101.3574                | 101.3574            | 101.357416534939          |
| 5   | 1      | h1tg000014l             | 130352633    | 120089316  | 108.5464                | 108.5464            | 108.546403078855          |
| 6   | 1      | h1tg000016l             | 113446105    | 117806340  | 96.2988                 | 96.2988             | 96.2988112524335          |
| 7   | 2      | h1tg000309l,h1tg000018l | 110888869    | 110682743  | 0.1765,100.0098         | 100.1863            | 100.186231380261          |
| 8   | 1      | h1tg000025l             | 124575867    | 113319770  | 109.933                 | 109.933             | 109.933039045173          |
| 9   | 1      | h1tg000013l             | 114130826    | 105454467  | 108.2276                | 108.2276            | 108.227587931387          |
| 10  | 1      | h1tg000034l             | 108155622    | 103308737  | 104.6917                | 104.6917            | 104.691650620025          |
| 11  | 1      | h1tg000005l             | 112201721    | 106982474  | 104.8786                | 104.8786            | 104.878600021906          |
| 12  | 1      | h1tg000006l             | 100320330    | 87216183   | 115.0249                | 115.0249            | 115.024903119184          |
| 13  | 1      | h1tg000011l             | 93858384     | 83472345   | 112.4425                | 112.4425            | 112.442490983091          |
| 14  | 1      | h1tg000001l             | 84104428     | 82403003   | 102.0648                | 102.0648            | 102.064760916541          |
| 15  | 1      | h1tg000004l             | 90708687     | 85007780   | 106.7063                | 106.7063            | 106.706335584814          |
| 16  | 1      | h1tg000019l             | 89667071     | 81013979   | 110.681                 | 110.681             | 110.680986302376          |
| 17  | 1      | h1tg000020l             | 82528804     | 73167244   | 112.7947                | 112.7947            | 112.794741865636          |
| 18  | 1      | h1tg000042l             | 72996406     | 65820629   | 110.902                 | 110.902             | 110.902018271506          |
| 19  | 2      | h1tg000031l,h1tg000026l | 64504278     | 63449741   | 69.8351,31.8269         | 101.662             | 101.662003632135          |
| 20  | 1      | h1tg000027l             | 74879700     | 71974595   | 104.0363                | 104.0363            | 104.036292250064          |
| 21  | 1      | h1tg000036l             | 78335069     | 69862954   | 112.1268                | 112.1268            | 112.126763205575          |
| 22  | 1      | h1tg000024l             | 77781670     | 60773035   | 127.9871                | 127.9871            | 127.987141007521          |
| 23  | 1      | h1tg000044l             | 58349949     | 52498615   | 111.1457                | 111.1457            | 111.145692129211          |
| 24  | 1      | h1tg000023l             | 63446359     | 62317253   | 101.8119                | 101.8119            | 101.811867413347          |
| 25  | 1      | h1tg000038l             | 48040529     | 42350435   | 113.4357                | 113.4357            | 113.435739207874          |
| 26  | 1      | h1tg000029l             | 60897474     | 51992305   | 117.1279                | 117.1279            | 117.127859593838          |
| 27  | 1      | h1tg000012l             | 46355633     | 45612108   | 101.6301                | 101.6301            | 101.630104445074          |
| 28  | 1      | h1tg000040l             | 43091862     | 45940150   | 93.8                    | 93.8                | 93.8000028297687          |
| 29  | 2      | h1tg000051l,h1tg000039l | 55379043     | 51098607   | 24.6007,83.7761         | 108.3768            | 108.376815438433          |

for T2T Wagyu and Charolais
** note for Wagyu, the query alignment filter is still at 5000bp hence the many small contigs (the others are set at >1Mb)

Male Wagyu

| chr | contig | query_name                                               | query_length | ref_length | contig_align_proportion | queryref_proportion | queryref_proportion_check |
|-----|--------|----------------------------------------------------------|--------------|------------|-------------------------|---------------------|---------------------------|
| 1   | 1      | haplotype2-0001614                                       | 173391082    | 158534110  | 109.3715                | 109.3715            | 109.371467124646          |
| 2   | 1      | haplotype2-0001615                                       | 156878617    | 136231102  | 115.1562                | 115.1562            | 115.156241634161          |
| 3   | 1      | haplotype2-0001639                                       | 136380939    | 121005158  | 112.7067                | 112.7067            | 112.706715361671          |
| 4   | 1      | haplotype2-0001658                                       | 126999771    | 120000601  | 105.8326                | 105.8326            | 105.832612454999          |
| 5   | 1      | haplotype2-0001629                                       | 137369161    | 120089316  | 114.3892                | 114.3892            | 114.389160980815          |
| 6   | 1      | haplotype2-0001641                                       | 113390177    | 117806340  | 96.2513                 | 96.2513             | 96.2513367277177          |
| 7   | 2      | haplotype2-0001645,haplotype2-0001766                    | 112325323    | 110682743  | 100.9284,0.5557         | 101.4841            | 101.484043452013          |
| 8   | 1      | haplotype2-0001633                                       | 129712032    | 113319770  | 114.4655                | 114.4655            | 114.465491767235          |
| 9   | 1      | haplotype2-0001638                                       | 123194416    | 105454467  | 116.8224                | 116.8224            | 116.822377946304          |
| 10  | 1      | haplotype2-0001643                                       | 112032441    | 103308737  | 108.4443                | 108.4443            | 108.444304183101          |
| 11  | 1      | haplotype2-0001620                                       | 115532783    | 106982474  | 107.9923                | 107.9923            | 107.992252076728          |
| 12  | 1      | haplotype2-0001631                                       | 110984222    | 87216183   | 127.2519                | 127.2519            | 127.251867924557          |
| 13  | 1      | haplotype2-0001647                                       | 102215726    | 83472345   | 122.4546                | 122.4546            | 122.454599783916          |
| 14  | 3      | haplotype2-0001617,haplotype2-0001741,haplotype2-0001619 | 101409327    | 82403003   | 38.1743,1.3079,83.5829  | 123.0651            | 123.065086596419          |
| 15  | 1      | haplotype2-0001636                                       | 86581406     | 85007780   | 101.8512                | 101.8512            | 101.851155270729          |
| 16  | 1      | haplotype2-0001624                                       | 96761639     | 81013979   | 119.4382                | 119.4382            | 119.438200906044          |
| 17  | 1      | haplotype2-0001632                                       | 82934539     | 73167244   | 113.3493                | 113.3493            | 113.349272797538          |
| 18  | 1      | haplotype2-0001630                                       | 81474408     | 65820629   | 123.7825                | 123.7825            | 123.782481628974          |
| 19  | 1      | haplotype2-0001653                                       | 72341849     | 63449741   | 114.0144                | 114.0144            | 114.014411816118          |
| 20  | 1      | haplotype2-0001640                                       | 84266000     | 71974595   | 117.0774                | 117.0774            | 117.077421554091          |
| 21  | 2      | haplotype2-0001652,haplotype2-0001635                    | 84372891     | 69862954   | 41.8464,78.9228         | 120.7692            | 120.76914325724           |
| 22  | 1      | haplotype2-0001648                                       | 80064047     | 60773035   | 131.7427                | 131.7427            | 131.742716156927          |
| 23  | 1      | haplotype2-0001650                                       | 66310779     | 52498615   | 126.3096                | 126.3096            | 126.309577881245          |
| 24  | 1      | haplotype2-0001646                                       | 72467140     | 62317253   | 116.2874                | 116.2874            | 116.287442901246          |
| 25  | 1      | haplotype2-0001656                                       | 78312078     | 42350435   | 184.9145                | 184.9145            | 184.914459556319          |
| 26  | 1      | haplotype2-0001644                                       | 58734772     | 51992305   | 112.9682                | 112.9682            | 112.968201736776          |
| 27  | 1      | haplotype2-0001657                                       | 63528600     | 45612108   | 139.2801                | 139.2801            | 139.280122725308          |
| 28  | 1      | haplotype2-0001642                                       | 43109272     | 45940150   | 93.8379                 | 93.8379             | 93.8378999633219          |
| 29  | 1      | haplotype2-0001651                                       | 56163682     | 51098607   | 109.9124                | 109.9124            | 109.912354362224          |


Charolais

| chr | contig | query_name                                                                  | query_length | ref_length | contig_align_proportion       | queryref_proportion | queryref_proportion_check |
|-----|--------|-----------------------------------------------------------------------------|--------------|------------|-------------------------------|---------------------|---------------------------|
| 1   | 1      | haplotype1-0000003                                                          | 182979407    | 158534110  | 115.4196                      | 115.4196            | 115.419581943596          |
| 2   | 2      | haplotype1-0000002,haplotype1-0000029                                       | 148816564    | 136231102  | 51.3882,57.8501               | 109.2383            | 109.238317693415          |
| 3   | 1      | haplotype1-0000036                                                          | 134103142    | 121005158  | 110.8243                      | 110.8243            | 110.824318745156          |
| 4   | 1      | haplotype1-0000011                                                          | 135334452    | 120000601  | 112.7781                      | 112.7781            | 112.778145169456          |
| 5   | 1      | haplotype1-0000026                                                          | 132583263    | 120089316  | 110.4039                      | 110.4039            | 110.40387889294           |
| 6   | 1      | haplotype1-0000020                                                          | 112746301    | 117806340  | 95.7048                       | 95.7048             | 95.7047821025592          |
| 7   | 4      | haplotype1-0000025,haplotype1-0000152,haplotype1-0000160,haplotype1-0000303 | 112505087    | 110682743  | 100.6217,0.4302,0.1983,0.3963 | 101.6465            | 101.646457207877          |
| 8   | 1      | haplotype1-0000016                                                          | 127333191    | 113319770  | 112.3663                      | 112.3663            | 112.366263186027          |
| 9   | 1      | haplotype1-0000022                                                          | 125518652    | 105454467  | 119.0264                      | 119.0264            | 119.026396482569          |
| 10  | 1      | haplotype1-0000023                                                          | 111123167    | 103308737  | 107.5642                      | 107.5642            | 107.564152100708          |
| 11  | 1      | haplotype1-0000007                                                          | 120347877    | 106982474  | 112.4931                      | 112.4931            | 112.493077137102          |
| 12  | 1      | haplotype1-0000032                                                          | 106642025    | 87216183   | 122.2732                      | 122.2732            | 122.273208172846          |
| 13  | 1      | haplotype1-0000031                                                          | 92427779     | 83472345   | 110.7286                      | 110.7286            | 110.728623953239          |
| 14  | 1      | haplotype1-0000017                                                          | 99710108     | 82403003   | 121.003                       | 121.003             | 121.00300276678           |
| 15  | 1      | haplotype1-0000019                                                          | 100568058    | 85007780   | 118.3045                      | 118.3045            | 118.304534008534          |
| 16  | 1      | haplotype1-0000006                                                          | 97054572     | 81013979   | 119.7998                      | 119.7998            | 119.799784182925          |
| 17  | 1      | haplotype1-0000014                                                          | 85817499     | 73167244   | 117.2895                      | 117.2895            | 117.289505943397          |
| 18  | 1      | haplotype1-0000033                                                          | 81841572     | 65820629   | 124.3403                      | 124.3403            | 124.340306744866          |
| 19  | 1      | haplotype1-0000030                                                          | 77457902     | 63449741   | 122.0776                      | 122.0776            | 122.077570025069          |
| 20  | 1      | haplotype1-0000018                                                          | 87652172     | 71974595   | 121.7821                      | 121.7821            | 121.782098252863          |
| 21  | 1      | haplotype1-0000039                                                          | 78270986     | 69862954   | 112.035                       | 112.035             | 112.035036480135          |
| 22  | 1      | haplotype1-0000008                                                          | 72488140     | 60773035   | 119.2768                      | 119.2768            | 119.27681413311           |
| 23  | 1      | haplotype1-0000034                                                          | 60268698     | 52498615   | 114.8005                      | 114.8005            | 114.800548547805          |
| 24  | 1      | haplotype1-0000028                                                          | 76159996     | 62317253   | 122.2133                      | 122.2133            | 122.21333953857           |
| 25  | 1      | haplotype1-0000040                                                          | 53108519     | 42350435   | 125.4025                      | 125.4025            | 125.402534826384          |
| 26  | 1      | haplotype1-0000024                                                          | 66656540     | 51992305   | 128.2046                      | 128.2046            | 128.204625665279          |
| 27  | 1      | haplotype1-0000041                                                          | 57848639     | 45612108   | 126.8274                      | 126.8274            | 126.827374433122          |
| 28  | 1      | haplotype1-0000021                                                          | 43144056     | 45940150   | 93.9136                       | 93.9136             | 93.9136158676017          |
| 29  | 1      | haplotype1-0000037                                                          | 55700141     | 51098607   | 109.0052                      | 109.0052            | 109.005204388448          |
